improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -653,16 +713,19 @@ def _probabilities_to_percentiles(
             raise NotImplementedError(msg)
 
         if not self.skip_ecc_bounds:
-            phenom_name = get_threshold_coord_name_from_probability_name(
-                forecast_probabilities.name()
-            )
-            cube_units = forecast_probabilities.coord(threshold_coord.name()).units
-            bounds_pairing = get_bounds_of_distribution(phenom_name, cube_units)
-            (threshold_points, probabilities_for_cdf) = (
-                self._add_bounds_to_thresholds_and_probabilities(
-                    threshold_points, probabilities_for_cdf, bounds_pairing
+            if self._assess_if_ecc_bounds_needed(
+                forecast_probabilities, threshold_points, threshold_name
+            ):
+                phenom_name = get_threshold_coord_name_from_probability_name(
```

Comment:
```suggestion
                threshold_name = get_threshold_coord_name_from_probability_name(
```
This is probably a clearer name - sorry for not pointing this out before.